### PR TITLE
Add unit tests for Postgres version check logic

### PR DESCRIPTION
**Pull request by gbonik**
[no release notes]

John requested unit tests for #1820

### CODE REVIEW

File: atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresTestSuite.java [hunk 1]
```diff
@@ -41,7 +42,8 @@
         DbkvsPostgresKeyValueServiceTest.class,
         DbkvsPostgresSerializableTransactionTest.class,
         DbkvsPostgresSweeperTest.class,
-        PostgresDbTimestampBoundStoreTest.class
+        PostgresDbTimestampBoundStoreTest.class,
+        PostgresVersionCheckTest.class
```
Comment (gsheasby) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresTestSuite.java]: This should not be in the integration test suite - it's a unit test file.

File: atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresVersionCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldLogErrorOn_9_1_24() {
+        Logger log = Mockito.mock(Logger.class);
+        PostgresVersionCheck.checkDatabaseVersion("9.1.24", log);
+        Mockito.verify(log).error(Mockito.argThat(new ArgumentMatcher<String>() {
+            @Override
+            public boolean matches(Object arg) {
+                return ((String)arg).contains("The minimum supported version is {}");
```
Comment (gsheasby) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresVersionCheckTest.java]: This line probably breaks checkstyle and findbugs. Please run locally (`atlasdb-dbkvs-tests:checkstyleTest` and `atlasdb-dbkvs-tests:findbugsTest`) and check the reports.

File: atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresTestSuite.java (hunk 1 above)
Comment (gbonik) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresTestSuite.java]: Yeah I thought about that, but didn't find a good existing suite to put it in. Do you have any suggestions? I'm not really familiar with your test infra -- do you scan the classpath for tests, or do you list the suites explicitly? I.e., if I leave that file on its own, will it run? Or do I have to make it part of a suite? And if there is no existing suite, how/where do I add it?

Comment (gsheasby) [inline atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsPostgresTestSuite.java]: I looked into it - you just needed to move the test to the test folder of `atlasdb-dbkvs`.

**Pull request merged**